resources/analysisTools/rnaseqworkflow/featureCountsDexseq_2_FpkmTpm

```diff
@@ -79,9 +79,18 @@ if (length $gtf_file > 0) {
     chomp $line;
     my @line_split = split ("\t", $line);
     if ($line_split[2] eq "gene") {
-      my @comment = split ("\"",$line_split[8]);
-      $gtf{$comment[1]}="$line_split[0]\t$line_split[3]\t$line_split[4]\t$comment[1]\t.\t$line_split[6]\t$comment[9]";
-      $gtf_names{$comment[1]}=$comment[9];
+      my @comment = split (";",$line_split[8]);
+      my ($gene_id, $gene_name);
+      for(my $i=0;$i<=$#comment;$i++){
+        if($comment[$i]=~/gene_id/){
+          ($gene_id)=$comment[$i]=~/\"(.*)\"/;
+        }
+        if($comment[$i]=~/gene_name/){
+          ($gene_name)=$comment[$i]=~/\"(.*)\"/;
```

Comment:
Can the value `$gene_name` be empty in the input file? If so this is fine, but if not the second question would be, what is the consequence, if the value **is** empty? Would that have negative consequences later in the code or in the result files? If yes, then you should add a check here and throw a descriptive error message. The same above.

One note: The GTF format is often incorrectly implemented and generated without quotes around attribute values! Maybe this is relevant here.

Reply:
The value `$gene_name` can be empty since it is not the primary index and its annotation columns could be empty. We are looking only at the gene annotations in the GTF file, so a properly formatted file should have a gene name in the gene row. 
For the GTF files downloaded from the gencode, the attributes are with quotes. I hope they will continue in this format, so I will leave it like this for now.